src/OrchardCore.Modules/OrchardCore.Liquid/Filters/DisplayUrlFilter.cs

```diff
@@ -18,21 +23,40 @@ public DisplayUrlFilter(IContentManager contentManager)
 
         public async ValueTask<FluidValue> ProcessAsync(FluidValue input, FilterArguments arguments, TemplateContext ctx)
         {
-            var contentItem = input.ToObjectValue() as ContentItem;
+            var isContentItemId = arguments["is_content_item_id"].Or(arguments.At(0)).ToBooleanValue();
+            RouteValueDictionary routeValues;
```

Comment:
You could do this without the `is_content_item_id`. turn the if around, and cast to `Contentitem` first. If that is null, then you can cast to string value and assume it is a content item id provided.